homeassistant/components/monoprice/__init__.py

```diff
@@ -57,11 +57,18 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
 async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     """Unload a config entry."""
     unload_ok = await hass.config_entries.async_unload_platforms(entry, PLATFORMS)
-    if unload_ok:
-        hass.data[DOMAIN][entry.entry_id][UNDO_UPDATE_LISTENER]()
+    if not unload_ok:
+        return False
+
+    hass.data[DOMAIN][entry.entry_id][UNDO_UPDATE_LISTENER]()
+
+    def _cleanup():
+        del hass.data[DOMAIN][entry.entry_id][MONOPRICE_OBJECT]
```

Comment:
Please add a comment explaining this is handed off to an executor because deleting the object has side effects.